app/src/components/dogs/DogForm.vue

```diff
@@ -0,0 +1,101 @@
+<template>
+ <form @submit.prevent="onSubmit(dog)">
+   <p>
+    <label> Name  
+      <input v-model="dog.name" required>
+    </label>
+   </p>
+   <p>
+    <label> Breed 
+      <input v-model="dog.breed" required>
+    </label> 
+  </p>
+  <p>
+    <label> Weight 
+      <input v-model="dog.weight" required> 
```

Comment:
```suggestion
      <input type="number" v-model.number="dog.weight" required> 
```
Type number adds form validation so the user can't enter strings (which will throw an error due to database schema/data types). v-model.number lets Vue know to typecast the data accordingly.